test/parallel/test-assert-async.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { promisify } = require('util');
+const wait = promisify(setTimeout);
+
+/* eslint-disable prefer-common-expectserror, no-restricted-properties */
+
+// Test assert.rejects() and assert.doesNotReject() by checking their
+// expected output and by verifying that they do not work sync
+
+assert.rejects(
+  () => assert.fail(),
+  common.expectsError({
+    code: 'ERR_ASSERTION',
+    type: assert.AssertionError,
+    message: 'Failed'
+  })
+);
+
+assert.doesNotReject(() => {});
+
+{
+  const promise = assert.rejects(async () => {
+    await wait(10);
```

Comment:
Just if you feel like it: our tests currently take to much time, so the smallest async operation would already be enough.